pkg/operator/osd/osd.go

```diff
@@ -230,11 +229,17 @@ func (c *Cluster) osdContainer(devices []Device, directories []Directory, select
 		envVars = append(envVars, locationEnvVar(config.Location))
 	}
 
+	// set the hostname to the host's name from the downstream api.
+	// the crush map doesn't like hostnames with periods, so we replace them with underscores.
+	hostnameUpdate := `echo $(HOSTNAME) | sed "s/\./_/g" > /etc/hostname; hostname -F /etc/hostname`
```

Comment:
does this only have impact on the pod itself, or does it actually affect the hostname of the node the pod has been scheduled on?